packages/config-manager/src/refresh.ts

```diff
@@ -0,0 +1,175 @@
+import type { Hash, OutPoint, Output, Script } from "@ckb-lumos/base";
+import type { ScriptConfig, ScriptConfigs } from "./types";
+import type { CKBComponents } from "@ckb-lumos/rpc/lib/types/api";
+
+type MaybePromise<T> = T | PromiseLike<T>;
+
+// prettier-ignore
+/**
+ * resolve the latest `OutPoint[]` that has consumed the old `OutPoint[]`
+ */
+export type ResolveLatestOutPointsOfTypeIds = (outPoints: OutPoint[]) => MaybePromise<OutPoint[]>;
+// prettier-ignore
+export type FetchOutputsByTxHashes = (txHashes: string[]) => MaybePromise<{ outputs: Output[] }[]>;
+// prettier-ignore
+/**
+ * fetch cells with corresponding type script
+ */
+export type FetchOutPointByTypeId = (scripts: Script[]) => MaybePromise<{ outPoint: OutPoint }[]>;
+
+// prettier-ignore
+/**
+ * the minimal batch RPC client
+ */
+export type BatchRequest = {
+  createBatchRequest<Params, Result>(params: Params[]): { exec(): Promise<Result[]> };
+};
+
+export function createRpcResolver(
+  rpc: BatchRequest
+): ResolveLatestOutPointsOfTypeIds {
+  const fetchTxs: FetchOutputsByTxHashes = async (txHashes) => {
+    const txs = await rpc
+      .createBatchRequest<
+        ["getTransaction", Hash],
+        CKBComponents.TransactionWithStatus
+      >(txHashes.map((txHash) => ["getTransaction", txHash]))
+      .exec();
+
+    return zipWith(txHashes, txs, (txHash, tx) => {
+      if (!tx?.transaction) {
+        throw new Error(`Cannot find transaction ${txHash}`);
+      }
+      return tx.transaction;
+    });
+  };
+
+  const fetchIndexerCells: FetchOutPointByTypeId = async (typeIds) => {
+    const res = await rpc
+      .createBatchRequest<unknown[], CKBComponents.GetLiveCellsResult<false>>(
+        typeIds.map((typeId) => [
+          "getCells",
+          {
+            script: typeId,
```

Comment:
Maybe the argument `typeId` could be the args in the script because other fields are determined. And a string is more like an `Id`, IMO.